FrescoImageViewFactory/src/main/java/com/github/piasy/biv/view/FrescoImageViewFactory.java

```diff
@@ -39,32 +39,46 @@
  */
 public class FrescoImageViewFactory extends ImageViewFactory {
     @Override
-    protected View createAnimatedImageView(final Context context, final int imageType,
-            final File imageFile, int initScaleType) {
-        SimpleDraweeView view = new SimpleDraweeView(context);
-        DraweeController controller = Fresco.newDraweeControllerBuilder()
-                .setUri(Uri.parse("file://" + imageFile.getAbsolutePath()))
-                .setAutoPlayAnimations(true)
-                .build();
-        view.setController(controller);
+    protected final View createAnimatedImageView(final Context context, final int imageType, int initScaleType) {
+        final SimpleDraweeView view = new SimpleDraweeView(context);
         view.getHierarchy().setActualImageScaleType(scaleType(initScaleType));
         return view;
     }
 
     @Override
-    public View createThumbnailView(final Context context, final Uri thumbnail,
-            final ImageView.ScaleType scaleType) {
-        SimpleDraweeView thumbnailView = new SimpleDraweeView(context);
-        DraweeController controller = Fresco.newDraweeControllerBuilder()
-                .setUri(thumbnail)
-                .build();
+    public final void loadAnimatedContent(final View view, final int imageType, final File imageFile) {
+        if (view instanceof SimpleDraweeView) {
+            final DraweeController controller = Fresco.newDraweeControllerBuilder()
+                    .setUri(Uri.parse("file://" + imageFile.getAbsolutePath()))
+                    .setAutoPlayAnimations(true)
+                    .build();
+            ((SimpleDraweeView) view).setController(controller);
+        }
+    }
+
+    @Override
+    public final View createThumbnailView(final Context context, final ImageView.ScaleType scaleType) {
+        final ImageView thumbnailView = new ImageView(context);
         if (scaleType != null) {
-            thumbnailView.getHierarchy().setActualImageScaleType(scaleType(scaleType));
+            thumbnailView.setScaleType(scaleType);
         }
-        thumbnailView.setController(controller);
         return thumbnailView;
+
+        /*final SimpleDraweeView thumbnailView = new SimpleDraweeView(context);
```

Comment:
Why comment them? If they are not needed, please delete them.

Reply:
forgot, removed! done!